README.md

```diff
@@ -26,55 +25,65 @@ We use [ISA-JSON](https://isatools.readthedocs.io/en/latest/isamodel.html) to st
-After a successful submission, BioSamples sends back a receipt containing BioSamples accession numbers for `Source` and `Sample` as `Source characteristics` and `Sample characteristics`, respectively. 
+2. **Identifying the Target Repositories**  
+The order of submission depends on the repositories specified in the ISA-JSON. MARS-CLI determines the correct sequence for submitting metadata and data.  
 
-    > The source code for the ISA-JSON API for BioSamples can be found in the [repository-services repo](/repository-services/isajson-biosamples/) and can be used for testing
+3. **Registering Samples in BioSamples**  
+MARS-CLI first submits the ISA-JSON to BioSamples via a newly developed API. BioSamples accessions are crucial since other repositories reuse them.  
 
-4. **Filtering the ISA-JSON**: The ISA-JSON (updated with BioSamples IDs) has to be filtered for every target repository so it only contains information relevant for that repo. This will be facilitated by the `target repository` attribute present in the ISA-JSON assays.
+- After submission, BioSamples returns a receipt containing accessions for `Source` and `Sample`, mapped to `Source characteristics` and `Sample characteristics`, respectively.  
+- Source Code: The BioSamples ISA-JSON API can be found in the [repository-services repo](/repository-services/isajson-biosamples/) and is available for testing.  
 
-5. **Submitting data to target repositories**: Since some repositories have the requirement that the actual data is already present in their upload space, the MARS-CLI could optionally take care of the data submission. This would guarantee the presence of the the data upon the metadata (ISA-JSON) submission and a checksum check.
+4. **Filtering the ISA-JSON**  
+Once updated with BioSamples accessions, the ISA-JSON is filtered for each target repository. This ensures that only relevant metadata is submitted to each repository. The filtering is based on the `target repository` attribute present in the ISA-JSON assays.  
 
-6. **Registering ISA-JSON at the target repositories**: Sending the filtered ISA-JSON to the endpoints of the repositories who accept ISA-JSON, for example ENA.
+5. **Submitting Data to Target Repositories**  
+MARS-CLI uploads data using FTP. Some repositories require that data files are present in their upload space before metadata submission. This step ensures that data availability and checksum validation are completed before sending metadata.  
 
-    > The source code for the ISA-JSON API for ENA can be found in the [repository-services repo](/repository-services/isajson-ena/) and can be used for testing
+6. **Registering ISA-JSON at the Target Repositories**  
+The filtered ISA-JSON is submitted to repositories that accept ISA-JSON metadata, such as ENA. The MARS project helps out with the adaptation of ISA-JSON by the repositories using the so called adapters. 
```

Comment:
Maybe link to the appropriate section for repository adapters - think it would be `#isa-json-support-by-repositories`

Reply:
Done!